reference/data-node-bootstrapping.md

```diff
@@ -0,0 +1,130 @@
+`add_data_node`.
+
+### Bootstrapping the database
+
+If `bootstrap` is true (which is the default), the access node will
+first connect to either the `postgres` database or, if the `postgres`
+database does not exist, to the `template0` database and create the
+database:
+
+```
+CREATE DATABASE <database>
+   TEMPLATE template0 OWNER <owner>
+   ENCODING <encoding> LC_COLLATE <collation>
+   LC_CTYPE <character type>
+```
+
+Here, the *owner* is the user connected on the access node, the
+*encoding*, *collation*, and *character type* is taken from the
+current database on the access node.
+
+Once the database is created, the access node disconnect and continue
+with bootstrapping the extension. If the database already exists on
+the data node, this step is skipped.
+
+[Currently, there is no validation of the database as there is if
+`bootstrap` is false. IMHO, we should add that. /Mats]
+
+### Validation of the database
+
+If `bootstrap` is false, the database will be validated by checking
+that the encoding, character type, and collation is the same as the
+current database on the access node.
+
+## Setting up the extension
+
+The `timescaledb` extension need to be installed in the database on
+the data node, so the next step is ensuring that there is a valid
+extension installed in the database by either creating the extension,
+or verifying that the extension is compatible with the extension
+installed on the access node.
+
+### Bootstrapping the extension
+
+#### Creating the schema for the extension
+
+To install the extension, a schema is created to store the objects
+that the `timescaledb` extension needs. The schema name is decided by
+picking the same schema name as on the access node and using that, so
+if you have created the `timescaledb` extension in the default manner,
+`public` will be used for the schema on the access node, hence
+`public` will be used on the data node as well.
+
+If the schema name for the extension on the access node is `public`,
+no new schema is created, but otherwise it is expected that the schema
+name does *not* exist.
+
+[I think we should change this and not create the schema if it exists,
+but otherwise proceed. /Mats]
+
+To change the schema used on the access node and data node, you can
+create the extension with the schema that you want:
+
+```
+CREATE SCHEMA magic;
+CREATE EXTENSION timescaledb SCHEMA magic;
+```
+
+When validating the schema, it is necessary that the schema is owned
+by the current user.
+
+> NOTE: This will place all the Timescale functions (for example,
+> `create_hypertable` in the `magic` schema), so you need to qualify
+> all function references with `magic`. For example:
+>
+> ```
+> SELECT * FROM magic.create_hypertable('conditions', 'time', 3);
+> ```
+
+#### Installing the extension
+
+Once there is a valid schema for the extension, the extension will be
+installed.
+
+[If the extension already exists, it is validated to ensure that it
```

Comment:
Should it be a comment that the correct version of the extension is verified?

Reply:
It is not currently verified since the bootstrapping fails if the extension already exists, but I think we should add it if we allow the extension to be pre-installed.

Reply:
You are right. It is checked when a connection is created, if I remember correctly.